src/App.tsx

```diff
@@ -1,7 +1,44 @@
 import React from 'react';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <p className="App__message">The last pressed key is [Enter]</p>
-  </div>
-);
+type State = {
+  pressedKey: string;
+};
+export class App extends React.Component<{}, State> {
+  state:State = {
+    pressedKey: '',
+  };
+
+  componentDidMount(): void {
+    document.addEventListener('keyup', this.handleKeyUp);
+  }
+
+  componentWillUnmount(): void {
+    document.removeEventListener('keyup', this.handleKeyUp);
```

Comment:
This will not remove event listener
https://dev.to/marcostreng/how-to-really-remove-eventlisteners-in-react-3och
https://bobbyhadz.com/blog/react-remove-event-listener

Reply:
Thx 😊  as I understand I forgot to bind this, so it could add and remove the same function.
```
 constructor(props){
        super(props);
        this.handleKeyUp = this.handleKeyUp.bind(this); //bind function once
    }
```
Anyway, I rewrote it to the functional component, from the recommendations above